### ISSUE TITLE
Required states can come from native elements [4e8ab6]

### ISSUE THREAD

**Issue by WilcoFiers** [auto-wcag/auto-wcag#2138]
This came from a comment from @JAWS-test on a [review](https://github.com/w3c/aria/discussions/1850#discussioncomment-4309723)

The following shouldn't fail the [Element with role attribute has required states and properties](https://www.w3.org/WAI/standards-guidelines/act/rules/4e8ab6/proposed/) rule. The checked state is present even without the aria-checked attribute, because it is on a native checkbox.

```html
<input type="checkbox" role="menuitemcheckbox">
```

**Comment by Jym77** [comment]
Decision:
* we need "implicitly set" and "explicitly set" definitions.
* "required state" rule should accept implicitly set states (i.e., this example is good, there is no author error)
* "allowed states" rule should only flagged explicitly set states (i.e. `<input type="checkbox" role="button">` shouldn't be flagged for un-allowed `aria-checked` since it is not an author error).
